### ISSUE TITLE
(KeyError) key {:pattern, "__key*__:mykey"} not found in: %{}

### ISSUE THREAD

**Issue by eppdot** [whatyouhide/redix_pubsub#13]
I receive this error message when I use `redix_pubsub`  and flush the whole redis - while testing.

```bash
13:46:42.724 [error] GenServer :redix_pubsub terminating
** (KeyError) key {:pattern, "__key*__:mykey"} not found in: %{}
    (elixir) lib/map.ex:255: Map.fetch!/2
    (redix_pubsub) lib/redix/pubsub/connection.ex:292: Redix.PubSub.Connection.handle_pubsub_msg/2
    (redix_pubsub) lib/redix/pubsub/connection.ex:179: Redix.PubSub.Connection.new_data/2
    (redix_pubsub) lib/redix/pubsub/connection.ex:110: Redix.PubSub.Connection.handle_info/2
    (connection) lib/connection.ex:810: Connection.handle_async/3
    (stdlib) gen_server.erl:601: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:667: :gen_server.handle_msg/5
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Last message: {:tcp, #Port<0.9712>, "*4\r\n$8\r\npmessage\r\n$22\r\n__key*__:mykey\r\n$28\r\n__keyspace@0__:feed:0:latest\r\n$3\r\nset\r\n"}
```

Line `connection.ex:292` expects that there is a `pattern` key inside the map, but in reality its empty. Maybe empty maps should be ignored?

**Comment by eppdot** [comment]
@whatyouhide ping

**Comment by whatyouhide** [comment]
Hey @kaelumania, I won’t have time for a while to look at this. Do you want to try and have a look yourself to see if something comes up? :)

**Comment by diasbruno** [comment]
@kaelumania Same here. In my case, sometimes `redix_pubsub` fails to register `{:channel, channel_name}` on its internal structure and redis is trying to send messages to it since it was subscribed.

@whatyouhide is it possible that redix_pubsub can have a race condition while subscribing?

**Comment by whatyouhide** [comment]
Released v0.5.0 which requires rearchitects a few things, so hopefully we won't see this again. If you see something similar, please reopen an issue.
